loadtest/control/simulcontroller/controller.go

```diff
@@ -48,103 +50,108 @@ func (c *SimulController) Run() {
 
 	c.status <- control.UserStatus{ControllerId: c.id, User: c.user, Info: "user started", Code: control.USER_STATUS_STARTED}
 
-	defer c.sendStopStatus()
-
-	if resp := control.SignUp(c.user); resp.Err != nil {
-		c.status <- c.newErrorStatus(resp.Err)
-	} else {
-		c.status <- c.newInfoStatus(resp.Info)
+	initActions := []userAction{
+		{
+			run: control.SignUp,
+		},
+		{
+			run: func(u user.User) control.UserActionResponse {
+				return c.login()
+			},
+		},
+		{
+			run: func(u user.User) control.UserActionResponse {
+				return c.joinTeam(u)
+			},
+		},
 	}
 
-	if resp := control.Login(c.user); resp.Err != nil {
-		c.status <- c.newErrorStatus(resp.Err)
-	} else {
-		c.status <- c.newInfoStatus(resp.Info)
-		c.connect()
-	}
+	defer func() {
```

Comment:
I would move the defer block on top of `initActions` declaration. So that the declaration and ranging of `initActions` remain close.

Reply:
Sounds good :+1: